core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingModalService.js

```diff
@@ -0,0 +1,109 @@
+                stateInteractionIdService, angularNameService) {
+              $scope.trainingDataAnswer = '';
+              $scope.trainingDataFeedback = '';
+              $scope.trainingDataOutcomeDest = '';
+
+              // See the training panel directive in StateEditor for an
+              // explanation on the structure of this object.
+              $scope.classification = {
+                answerGroupIndex: 0,
+                newOutcome: null
+              };
+
+              $scope.finishTraining = function() {
+                $modalInstance.close();
+              };
+
+              $scope.init = function() {
+                var explorationId =
+                  explorationContextService.getExplorationId();
+                var currentStateName =
+                  editorContextService.getActiveStateName();
+                var state = explorationStatesService.getState(currentStateName);
+
+                // Retrieve the interaction ID.
+                var interactionId = stateInteractionIdService.savedMemento;
+
+                var rulesServiceName =
+                  angularNameService.getNameOfInteractionRulesService(
+                    interactionId)
+
+                // Inject RulesService dynamically.
+                var rulesService = $injector.get(rulesServiceName);
+
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    explorationId, currentStateName, state, unhandledAnswer,
+                    true, rulesService));
+                var feedback = 'Nothing';
+                var dest = classificationResult.outcome.dest;
+                if (classificationResult.outcome.feedback.length > 0) {
+                  feedback = classificationResult.outcome.feedback[0];
+                }
+                if (dest === currentStateName) {
+                  dest = '<em>(try again)</em>';
+                }
+
+                // $scope.trainingDataAnswer, $scope.trainingDataFeedback
+                // $scope.trainingDataOutcomeDest are intended to be local
+                // to this modal and should not be used to populate any
+                // information in the active exploration (including the
+                // feedback). The feedback here refers to a representation
+                // of the outcome of an answer group, rather than the
+                // specific feedback of the outcome (for instance, it
+                // includes the destination state within the feedback).
+                $scope.trainingDataAnswer = unhandledAnswer;
+                $scope.trainingDataFeedback = feedback;
+                $scope.trainingDataOutcomeDest = dest;
+                $scope.classification.answerGroupIndex = (
+                  classificationResult.answerGroupIndex);
+              };
+
+              $scope.init();
+            }]
+        });
+      }
+    };
+  }
+]);
+
```

Comment:
Ditto.

Reply:
@seanlip  , i did this in my second last commit , then too it showed the same error . 
Here's the link : - https://github.com/oppia/oppia/pull/3913/commits/f51bf86d36ee57c608779fec489f5ee4b2ef4fd3

Reply:
If you're using Sublime, btw, there's a user setting that might help:

     `"ensure_newline_at_eof_on_save": true,`